core/src/main/scala/org/broadinstitute/dsde/rawls/entities/EntityManager.scala

```diff
@@ -64,10 +64,13 @@ class EntityManager(providerBuilders: Set[EntityProviderBuilder[_ <: EntityProvi
     * @return
     */
   def resolveProviderFuture(entityRequestArguments: EntityRequestArguments)(implicit executionContext: ExecutionContext): Future[EntityProvider] = {
-    Future.fromTry(resolveProvider(entityRequestArguments)).recoverWith {
-      case regrets: DataEntityException =>
-        Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, regrets)))
-    }
+    // TODO: (DA) the recoverWith here seems to make error-handling more complicated as it swallows and rewrites
+    // all DataEntityExceptions, including adding big stack traces to it. Why does it exist?
+    Future.fromTry(resolveProvider(entityRequestArguments))
+//      .recoverWith {
+//        case regrets: DataEntityException =>
+//          Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, regrets)))
+//      }
```

Comment:
need to either remove the commented-out code or reinstate it; we shouldn't merge as-is

Reply:
im cool with removing it

Reply:
turns out it's useful; I've rewritten it to do what I expected it to do (propagate the status code)